pwiz_tools/Skyline/Model/SmallMoleculeTransitionListReader.cs

```diff
@@ -1467,6 +1419,156 @@ public static string GetAcceptedIonMobilityUnitsString()
             return null;
         }
 
+        private bool ProcessNeutralLoss(Row row, int indexNeutralLoss, ref string formula)
+        {
+            // Derive product formula from neutral loss?
+            var neutralLoss = GetCellTrimmed(row, indexNeutralLoss);
+            if (!string.IsNullOrEmpty(neutralLoss))
+            {
+
+                var precursorFormula = Adduct.SplitFormulaAndTrailingAdduct(GetCellTrimmed(row, INDEX_MOLECULE_FORMULA), Adduct.ADDUCT_TYPE.charge_only, out _); // Removes any adduct description e.g. C12H5[M+H] => C12H5
+                if (string.IsNullOrEmpty(precursorFormula))
+                {
+                    // There's no use for a loss formula if there's no precursor formula
+                    ShowTransitionError(new PasteError
+                    {
+                        Column = indexNeutralLoss,
+                        Line = row.Index,
+                        Message = Resources.SmallMoleculeTransitionListReader_ProcessNeutralLoss_Cannot_use_product_neutral_loss_chemical_formula_without_a_precursor_chemical_formula
+                    });
+                    return false;
+                }
+
+                // Get neutral loss formula
+                try
+                {
+                    BioMassCalc.MONOISOTOPIC.CalculateMassFromFormula(neutralLoss); // For syntax check - throws ArgumentException if there's a problem

```

Comment:
I still don't get what is the extra checking beyond Molecule.Parse() below. It still feels like this should not be necessary and the fact that you need the comment indicates the use and result are unexpected.